src/pages/index.js

```diff
@@ -20,58 +33,205 @@ const enableValidation = (settings) => {
+ userinfo.setUserInfo(formData);
+ userinfo.setAvatar(formData.avatar);
+ cardData.forEach((item) => {
+ 
+  //console.log("is like and unlike here at ForEach?",cardData)
+  //cardData.setLikeStatus(cardData.isLiked)
+  //const cardElement = createCard(item)
+   section.addItem(createCard(item));
+   //console.log("is like and unlike here at addItem???")
+ })
+})
+.catch((err) => {
+ console.error(err);// log the error to the console
+})
+.finally(() => {
+section.renderItems();
+//console.log("is like and unlike here at renderItems???", renderItems)
+}); 
+
 //Popups
+
+//PopUpWithImage
 const popUpWithImage = new PopUpWithImage("#modal-previewImage");
 popUpWithImage.setEventListeners();
 
+//PopUpWithForm
 const popUpEditProfile = new PopUpWithForm(
   "#modal-edit-profile",
-  (formData) => {
-    userinfo.setUserInfo(formData);
-    popUpEditProfile.close();
+  ({name, about}) => {
+    return new Promise((resolve, reject) => {
+      api.updateEditProfile({name, about})
+      .then((updateEditProfile) => {
+        userinfo.setUserInfo(updateEditProfile);
+        popUpEditProfile.close();
+        resolve();
+      })
+      .catch((err) => {
+        console.error(err);
+        reject(err);
+      })
+    })
   }
 );
 popUpEditProfile.setEventListeners();
 
+//PopUpWithForm
 const popUpAddItem = new PopUpWithForm("#modal-add-profile", (formData) => {
-  const name = formData.name;
-  const link = formData.link;
-  handleAddProfileFormSubmit(name, link);
+  return new Promise ((resolve, reject) => {
+    const name = formData.name;
+    const link = formData.link;
+    handleAddProfileFormSubmit(name, link)
+    .then(() => {
+      popUpAddItem.close();
+      resolve();
+  })
+  .catch((err) => {
+    console.error(err);
+    reject(err);
+   })
+  })
 });
 popUpAddItem.setEventListeners();
 
+//PopUpWithForm
+const popUpAvatar = new PopUpWithForm("#modal-avatar", (formData) => {
+  //const avatarLink = cardData.avatar;
+  const avatar = formData.avatar;
+  return new Promise ((resolve, reject) => {
+    api.updateAvatar(avatar)
+    .then((updatedAvatar) => {
+      userinfo.setAvatar(updatedAvatar.avatar);
+      popUpAvatar.close();
+      resolve();
+    })
+    .catch((err) => {
+      console.error(err);
+      reject(err);
+    })
+  })
+});
+popUpAvatar.setEventListeners();
+
+//PopUpWithConfirmation
+const popUpConfirm = new PopUpWithConfirmation("#modal-confirm-delete")
+popUpConfirm.setEventListeners();
+
+
 //Event Handlers
 function handleAddProfileFormSubmit(title, link) {
-  const newCard = createCard({ name: title, link });
-  section.addItem(newCard);
+  return new Promise((resolve, reject) => {
+  api.createCard({ name: title, link })
+  .then((card) => {
+    section.addItem(createCard(card));
+    console.log("CARD CHECK! please please work T_T.", card);
+    resolve(); 
+  })
+.catch((err) => {
+  console.error(err);
+  reject(err);
+}) 
+.finally(() => {
   popUpAddItem.close();
```

Comment:
`        popUpAddItem.close();
`

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well